modules/new_profile/php/new_profile.class.inc

```diff
@@ -97,207 +54,82 @@ class New_Profile extends \NDB_Form
-                       'Female' => 'Female',
-                      );
-        $this->addSelect('sex', 'Sex', $sexOptions);
-
-        // sex rules
-        $this->addRule('sex', 'Sex is required', 'required');
-
-        // add sites for users with more than one site affiliation
-        $user =& \User::singleton();
-        $DB   = \Database::singleton();
+        $DB     = \Database::singleton();
+        $config = \NDB_Config::singleton();
+        $user   = \User::singleton();
+        $this->result['startYear']  = $config->getSetting('startYear');
+        $this->result['endYear']    = $config->getSetting('endYear');
+        $this->result['ageMax']     = $config->getSetting('ageMax');
+        $this->result['ageMin']     = $config->getSetting('ageMin');
+        $this->result['dobFormat']  = $config->getSetting('dobFormat');
+        $this->result['edc']        = $config->getSetting('useEDC');
+        $this->result['useProject'] = $config->getSetting('useProjects');
+        $this->result['sex']        = array(
+                                       'male'   => 'Male',
+                                       'female' => 'Female',
+                                      );
+        $this->result['pscidSet']   = "false";
+        //get sites for the select dropdown
         $user_list_of_sites = $user->getData('CenterIDs');
         $num_sites          = count($user_list_of_sites);
-
-        if ($num_sites >1) {
-            $psc_labelOptions = array(null => '');
+        if ($num_sites >=1) {
             foreach ($user_list_of_sites as $key => $siteID) {
                 $center = $DB->pselectRow(
                     "SELECT CenterID as ID, Name FROM psc WHERE CenterID =:cid",
                     array('cid' => $siteID)
                 );
                 $psc_labelOptions[$siteID] = $center['Name'];
             }
-            $this->addSelect('psc', 'Site', $psc_labelOptions);
-
-            // site rules
-            $this->addRule('psc', 'Site is required', 'required');
+        } else {
+            $psc_labelOptions = array(null => '');
         }
-
+         $this->result['site'] = $psc_labelOptions;
+         //get projects for the select dropdown
+        $projects = \Utility::getProjectList();
+        foreach ($projects as $key=>$value) {
+             $list_of_projects[$key] = $value;
+        }
+        $this->result['project'] = $list_of_projects;
+         //get setting through pscid
         $PSCIDsettings = $config->getSetting('PSCID');
         if ($PSCIDsettings['generation'] == 'user') {
-            $this->addBasicText('PSCID', 'PSCID');
+            $result['pscidSet'] = "true";
         }
-
-        $this->form->addFormRule(array(&$this, '_validate'));
+        return true;
     }
     /**
-     * Validate function
+    * Call toJSON to display a json file on the page.
+    *
+    * @return string a json
+    */
+    function display()
+    {
+        return $this->toJSON();
```

Comment:
the `result` array is getting printed on the page before the React elements have time to render. is this necessary?